src/main/java/net/cardosi/mojo/cache/CachedProject.java

```diff
@@ -315,6 +289,21 @@ public void run() {
         });
     }
 
+    public WatchService createWatchService() throws IOException {
+        boolean isMac = System.getProperty("os.name").toLowerCase().contains("mac");
+        if (isMac) {
+            return new MacOSXListeningWatchService(new MacOSXListeningWatchService.Config() {});
+        } else {
+            return FileSystems.getDefault().newWatchService();
+        }
+    }
+
+    private void registerWatch(Path stepDirPath, WatchService w) throws IOException {
+        boolean isMac = System.getProperty("os.name").toLowerCase().contains("mac");
+        Watchable watchable = isMac ? new WatchablePath(stepDirPath) : stepDirPath;
```

Comment:
Extract a helper called isMac ?

Reply:
or perhaps a static final boolean so we only compute it once? Or a helper factory/instance thingie that provides both this and the above?

Reply:
done